rs/execution_environment/src/execution/update.rs

```diff
@@ -341,26 +359,22 @@ impl UpdateHelper {
 
         validate_message(&canister, &original.method)?;
 
-        if let CanisterCallOrTask::Call(_) = original.call_or_task {
-            // TODO(RUN-957): Enforce the limit in heartbeat and timer after
-            // canister logging ships by removing the `if` above.
+        let wasm_memory_usage = canister
+            .execution_state
+            .as_ref()
+            .map_or(NumBytes::new(0), |es| {
+                num_bytes_try_from(es.wasm_memory.size).unwrap()
+            });
 
-            let wasm_memory_usage = canister
-                .execution_state
-                .as_ref()
-                .map_or(NumBytes::new(0), |es| {
-                    num_bytes_try_from(es.wasm_memory.size).unwrap()
-                });
+        if let Some(wasm_memory_limit) = clean_canister.system_state.wasm_memory_limit {
+            // A Wasm memory limit of 0 means unlimited.
+            if wasm_memory_limit.get() != 0 && wasm_memory_usage > wasm_memory_limit {
+                let err = HypervisorError::WasmMemoryLimitExceeded {
+                    bytes: wasm_memory_usage,
+                    limit: wasm_memory_limit,
+                };
 
-            if let Some(wasm_memory_limit) = clean_canister.system_state.wasm_memory_limit {
-                // A Wasm memory limit of 0 means unlimited.
-                if wasm_memory_limit.get() != 0 && wasm_memory_usage > wasm_memory_limit {
-                    let err = HypervisorError::WasmMemoryLimitExceeded {
-                        bytes: wasm_memory_usage,
-                        limit: wasm_memory_limit,
-                    };
-                    return Err(err.into_user_error(&canister.canister_id()));
-                }
+                return Err(err.into_user_error(&canister.canister_id()));
```

Comment:
The only change here is that `if` statement
`if let CanisterCallOrTask::Call(_) = original.call_or_task {`
is deleted, as previously discussed in the private group, that should be safe to do.